src/solver/conv_MP_bidirectional_winograd.cpp

```diff
@@ -0,0 +1,536 @@
+#endif
+        if(limit != std::numeric_limits<std::size_t>::max())
+        {
+            const auto required = GetWorkspaceSize(params);
+            MIOPEN_LOG_I2("Workspace required: " << required << ", limit: " << limit);
+            if(required > limit)
+                return false;
+        }
+    }
+
+    // int offset for Workspace buffers.
+    if(((GetWinoBuffer<WinoDataH, WinoFilterH, WinoDataW, WinoFilterW>(params,
+                                                                       ConvWinoBuffType::Input))
+                .buff_info.total_byte_size /
+            GetTypeSize(params.in_data_type) +
+        (GetWinoBuffer<WinoDataH, WinoFilterH, WinoDataW, WinoFilterW>(params,
+                                                                       ConvWinoBuffType::Output))
+                .buff_info.total_byte_size /
+            GetTypeSize(params.in_data_type)) >= (1LL << 31))
+    {
+        return false;
+    }
+
+    assert(params.weights_layout.length() == 0); // _weights_layout is not supported yet
+
+    {
+        unsigned int const waves_in_group = 512 / wave_size;
+        unsigned int const tiles_per_wave = 8;
+        auto const tiles_per_group        = waves_in_group * tiles_per_wave / 2;
+        auto const n_groups               = params.GetStream().GetMaxComputeUnits();
+        auto const tiles_step             = tiles_per_group * n_groups;
+        if(tiles_step >= std::pow(2, 16))
+            return false;
+    }
+    {
+        unsigned long long const f_tiles_hw =
+            Ceil(params.kernel_size_w, WinoFilterW) * Ceil(params.kernel_size_h, WinoFilterH);
+        unsigned long long const d_tiles_hw =
+            Ceil(params.out_width, WinoDataW) * Ceil(params.out_height, WinoDataH);
+        unsigned long long const G_N_C_xTile =
+            params.n_inputs * params.group_counts * params.batch_sz * d_tiles_hw;
+        unsigned long long const G_K_C_xTile =
+            params.n_inputs * params.group_counts * params.n_outputs * f_tiles_hw;
+        unsigned long long const G_K_N_xTile =
+            params.batch_sz * params.group_counts * params.n_outputs * d_tiles_hw;
+        if(G_N_C_xTile >= std::pow(2, 24) || G_K_C_xTile >= std::pow(2, 24) ||
+           G_K_N_xTile >= std::pow(2, 24))
+            return false;
+    }
+
+    // clang-format off
+    bool ok = (
+           (params.kernel_size_w == WinoFilterW
+               && params.kernel_size_h == WinoFilterH)
+        && (params.kernel_stride_w == 1)
+        && params.kernel_stride_h == params.kernel_stride_w
+        && params.kernel_dilation_w == 1
+        && params.kernel_dilation_h == 1
+        && params.batch_sz < std::pow(2, 16)
+        && params.n_inputs < std::pow(2, 16)
+        && params.n_outputs < std::pow(2, 16)
+        && params.out_height < std::pow(2, 16)
+        && params.out_width < std::pow(2, 16)
+        && params.bias == 0
+        && params.in_layout == "NCHW");
```

Comment:
Are there checks for input dilation and output stride?

Reply:
&& (params.kernel_stride_w == 1)
        && params.kernel_stride_h == params.kernel_stride_w
        && params.kernel_dilation_w == 1
        && params.kernel_dilation_h == 1